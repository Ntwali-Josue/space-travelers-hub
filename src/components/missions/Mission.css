.mission {
  width: 10%;
}

.desc {
  width: 60%;
}

.status {
  width: 15%;
}

.action {
  width: 15%;
}
