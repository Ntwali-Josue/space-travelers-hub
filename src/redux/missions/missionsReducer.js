import { createAsyncThunk } from '@reduxjs/toolkit';
import fetchAPI from '../fetchAPI';
// import axios from 'axios'

const FETCH_MISSIONS = 'spaceX/missions/FETCH_MISSIONS/fulfilled';
const JOIN_MISSION = 'spaceX/missions/FETCH_MISSIONS';

const initialState = {
  status: 'empty',
  missionList: [],
  joinMission: [],
};

export const fetchMission = createAsyncThunk('spaceX/missions/FETCH_MISSIONS', async () => {
  const result = await fetchAPI('https://api.spacexdata.com/v3/missions');
  return result;
});

export const joinMisssion = (payload) => ({
  type: JOIN_MISSION,
  payload,
});

const missionsReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_MISSIONS:
      return {
        status: 'fetched',
        missionList: [...state.missionList, ...action.payload],
      };
    case JOIN_MISSION:
      return {
        status: state.status,
        missionList: state.missionList,
        joinMission: [...state.joinMission, action.payload],
      };

    default:
      return state;
  }
};

export default missionsReducer;
