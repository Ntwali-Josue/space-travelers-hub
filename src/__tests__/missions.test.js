import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import missionsReducer from '../redux/missions/missionsReducer';
import fetchAPI from '../redux/fetchAPI';
import Mission from '../components/missions/Missions';

describe('Mission reducer', () => {
  test('initial state', () => {
    expect(missionsReducer(undefined, {})).toEqual(
      {
        status: 'empty',
        missionList: [],
        joinedMission: [],
      },
    );
  });

  test('fetch mission action', () => {
    const initialState = {
      status: 'empty',
      missionList: [],
      joinedMission: [],
    };

    const action = {
      type: 'spaceX/missions/FETCH_MISSIONS/fulfilled',
      payload: [{ name: 'Thaicom' }, { name: 'Iridium NEXT' }],
    };

    expect(missionsReducer(initialState, action)).toEqual(
      {
        status: 'fetched',
        missionList: action.payload,
        joinedMission: [],
      },
    );
  });

  test('render missions', async () => {
    const missionList = await fetchAPI('https://api.spacexdata.com/v3/missions');
    const mapMissions = missionList.map(
      (mission) => (
        <Mission
          key={mission.mission_id}
          id={mission.mission_id}
          missionName={mission.mission_name}
          description={mission.description}
        />
      ),
    );
    render(
      <Provider store={store}>
        {mapMissions}
      </Provider>,
    );
    expect(screen.getByText('Thaicom')).toMatchSnapshot();
    expect(screen.getByText('Iridium NEXT')).toMatchSnapshot();
    expect(screen.getByText('Telstar')).toMatchSnapshot();
    expect(screen.getByText('SES')).toMatchSnapshot();
  });
});
