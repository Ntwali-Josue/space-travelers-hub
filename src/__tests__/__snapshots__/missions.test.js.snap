// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission reducer render missions 1`] = `
<td
  class="mission-names"
>
  Thaicom
</td>
`;

exports[`Mission reducer render missions 2`] = `
<td
  class="mission-names"
>
  Iridium NEXT
</td>
`;

exports[`Mission reducer render missions 3`] = `
<td
  class="mission-names"
>
  Telstar
</td>
`;

exports[`Mission reducer render missions 4`] = `
<td
  class="mission-names"
>
  SES
</td>
`;
